### PR TITLE
put alias error

### PR DESCRIPTION
**Pull request by berkin**
the push command also run with the parameter and it gives error.
```sh
git put test 
[master b9f5249] test
 1 file changed, 1 insertion(+), 1 deletion(-)
fatal: 'test' does not appear to be a git repository
fatal: Could not read from remote repository.
```

### CODE REVIEW

File: gitalias.txt [hunk 1]
```diff
@@ -505,7 +505,7 @@
 
   # Do everything we can to synchronize all changes
   get = !git pull --rebase && git submodule update --init --recursive
-  put = !git commit --all --message="$1" && git push
+  put = "!f() { git add -A && git commit -m \"$@\" && git push;  }; f"
```
Comment (phdru) [inline gitalias.txt]: There is a big difference between `git commit --all` and `git add -A && git commit`. Is the change intentional?

Comment (phdru) [inline gitalias.txt]: Also can I ask to fix minor style problem: two spaces before `}`?

**Comment by joelparkerhenderson** [comment]
Good discovery. Thank you. Definitely a bug.

The intent of `get` and `put` is primarily "Do the typical thing for the typical case".

What are opinions on whether adding the "git add -A" is the typical thing for the typical case? 

If so, this would alter the alias, beyond just fixing the bug.

Just doing a bug fix could be more akin to this:

    git commit --all --message="$1"; shift; git push \"$@\"

**Comment by erikmd** [comment]
> What are opinions on whether adding the "git add -A" is the typical thing for the typical case?

I think `git add -A && git commit -m ... && git push` is probably too rude, because *it will silently add everything that is not untracked, then commit and push it*... so I'd vote for keeping this PR as a bugfix and just run

> `git commit --all --message="$1"; shift; git push \"$@\" `

File: gitalias.txt (hunk 1 above)
Comment (joelparkerhenderson) [reply gitalias.txt]: Excellent point, thank you.

**Comment by erikmd** [comment]
Another solution for accepting multiple arguments would be to turn them into several `-m` flags, i.e.:

`git commit -m "Title" -m "First paragraph..." -m "Second paragraph..."`

File: gitalias.txt (hunk 1 above)
Comment (berkin) [reply gitalias.txt]: thanks, i fixed.

**Comment by berkin** [comment]
> What are opinions on whether adding the "git add -A" is the typical thing for the typical case?

The comment above the aliases says it will synchronise all changes so adding untracked files makes sense to me.

**Comment by joelparkerhenderson** [comment]
Thanks all for comments!

Here's feedback I received from a peer about principles: 

  * Fix each bug in its own commit. 
  * Do the fix with similar syntax if possible to make it easier for readers to understand.
  * Do any implementation change (e.g. change to `f()`) in its own commit as a "Refactor ..."
  * Do any feature change (e.g. change to `add --all`) in its own commit as an "Add ..."

Here's the fix code that seems to work for me, that's the closest to the syntax:

     put = !git commit --all --message=\"$1\" && shift && git push

@berkin Do you want credit for this fix? If you do, can you adjust the PR to be just the fix? If you don't care, then let me know and I can do it. If you would like, you can do a subsequent PR to do the refactor of `get` and `put` to use functions.

Thanks!

**Pull request merged**